### PR TITLE
Loadingspinner

### PR DESCRIPTION
**Pull request by janhenrikfock**
I have implemented a loading-spinner.gif in a component to bridge long loadingtimes at initial load of the page.

The docs to check: 
Component -> LoadingAnimation.js
Implementation in Component: AllMonsters.js

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+__snapshots__
```
Comment (karstenlenz) [inline .gitignore]: Are you sure you don't need the snapshots in the repo? Without snapshots your tests won't work if you clone the repo.

File: src/components/LoadingAnimation.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+
+import LoadingAnimation from './LoadingAnimation'
+
+describe('LoadingAnimation', () => {
+  it('displays loadingspinner gif', () => {
+    const { getByAltText } = render(<LoadingAnimation alt={'loading'} />)
```
Comment (karstenlenz) [inline src/components/LoadingAnimation.test.js]: I don't think you need the curly braces around {'loading'} since it's just a string.

**Review by karstenlenz — APPROVED** [review]
Looks great! Just added 2 tiny comments.

File: src/components/LoadingAnimation.test.js (hunk 2 above)
Comment (janhenrikfock) [reply src/components/LoadingAnimation.test.js]: Just tested. You are correct, the curly-braces aren't necessary.

File: src/components/AllMonsters.js [hunk 3]
```diff
@@ -2,16 +2,21 @@ import React, { useEffect, useState } from 'react'
...
 export default function AllMonsters() {
   const [monsterDetails, setMonsterDetails] = useState([])
 
+  const [loading, setLoading] = useState(false)
+
   useEffect(() => {
+    localStorage.clear()
     const locallyLoadedMonsterNames = JSON.parse(
       localStorage.getItem('monsternames')
     )
```
Comment (mpagels) [inline src/components/AllMonsters.js]: if you clear the localStorage locallyLoadedMonsterNames will be always be null. So you don't need the JSON.parse
```suggestion
    localStorage.clear()
    const locallyLoadedMonsterNames = null
    )
```

File: src/components/LoadingAnimation.stories.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// import React from 'react'
+// import LoadingAnimation from './LoadingAnimation'
+// import loadingspinner from './images/loadingspinner.gif'
+
+// export default {
+//   title: 'LoadingAnimation',
+//   component: LoadingAnimation,
+// }
+
+// export const LoadingAnimation = <img src={loadingspinner} alt="loading" />
```
Comment (mpagels) [inline src/components/LoadingAnimation.stories.js]: file could be removed from project

File: src/components/AllMonsters.js [hunk 5]
```diff
@@ -33,15 +39,21 @@ export default function AllMonsters() {
...
+  } else {
+    return (
+      <>
+        <HeadlineStyled>Monsters of DnD</HeadlineStyled>
+        {monsterDetails.map((monster) => (
+          <OneMonsterItem key={monster.id} monster={monster}></OneMonsterItem>
+        ))}
+      </>
+    )
+  }
```
Comment (mpagels) [inline src/components/AllMonsters.js:53]: try to use ternary operator instead of if else?

**Review by mpagels — COMMENTED** [review]
is it intended that you want to delete the localStorage every time?
pls review my suggestions.

File: src/components/AllMonsters.js (hunk 3 above)
Comment (janhenrikfock) [reply src/components/AllMonsters.js]: good spot. That was for testing purposes. Thanks

**Pull request merged**